msgraph-mail/typescript/src/models/followupFlagImpl.ts

```diff
@@ -0,0 +1,62 @@
+import {createDateTimeTimeZoneFromDiscriminatorValue} from './createDateTimeTimeZoneFromDiscriminatorValue';
+import {DateTimeTimeZone} from './dateTimeTimeZone';
+import {FollowupFlag} from './followupFlag';
+import {FollowupFlagStatus} from './followupFlagStatus';
+import {DateTimeTimeZoneImpl} from './index';
+import {AdditionalDataHolder, Parsable, ParseNode, SerializationWriter} from '@microsoft/kiota-abstractions';
+
+export class FollowupFlagImpl implements FollowupFlag {
+    /** Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well. */
+    public additionalData: Record<string, unknown>;
+    /** The completedDateTime property */
+    public completedDateTime?: DateTimeTimeZone | undefined;
+    /** The dueDateTime property */
+    public dueDateTime?: DateTimeTimeZone | undefined;
+    /** The flagStatus property */
+    public flagStatus?: FollowupFlagStatus | undefined;
+    /** The startDateTime property */
+    public startDateTime?: DateTimeTimeZone | undefined;
+    /**
+     * Instantiates a new followupFlag and sets the default values.
+     * @param followupFlagParameterValue 
+     */
+    public constructor(followupFlagParameterValue?: FollowupFlag | undefined) {
+        this.additionalData = followupFlagParameterValue?.additionalData ? followupFlagParameterValue?.additionalData! : {};
+        this.completedDateTime = followupFlagParameterValue?.completedDateTime;
+        this.dueDateTime = followupFlagParameterValue?.dueDateTime;
```

Comment:
here I believe we should call the date time time zone constructor to cascade the behaviour as well